### PR TITLE
Feat: upgraded key system

### PR DESCRIPTION
**Pull request by SupremoUGH**
Second part of #153. It consists of the changes related to the key system. 

Changes:
`manta-accounting`: 
- `key.rs`: Upgrade to the new key system. Deprecated `HierarchicalKeyDerivationScheme` in favour of `AccountCollection`.
- `signer.rs`: Minimal changes so it compiles.

`manta-pay`:
- `key.rs`: Upgrade to the new system. Now we don't use key pairs, just keys.
- `signer::base`: Implemented `AccountCollection` and `DeriveAddresses` for `KeySecret`. 
- `signer`: minimal changes so it compiles.

---

Before we can merge this PR, please make sure that all the following items have been checked off:

- [x] Linked to an issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** line describing your change in [`CHANGELOG.md`](https://github.com/manta-network/manta-rs/blob/main/CHANGELOG.md) and added the appropriate `changelog` label to the PR.
- [x] Re-reviewed `Files changed` in the GitHub PR explorer.
- [x] Checked that changes and commits conform to the standards outlined in [`CONTRIBUTING.md`](https://github.com/manta-network/manta-rs/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by bhgomes — CHANGES_REQUESTED** [review]
We need `bip0039` for the signer changes. @AlexParshh please check this PR in the key system

File: manta-pay/src/key.rs [hunk 1]
```diff
@@ -240,28 +233,26 @@ where
 pub struct Mnemonic(
     /// Underlying BIP39 Mnemonic
     #[cfg_attr(feature = "serde", serde(serialize_with = "Mnemonic::serialize"))]
-    bip0039::Mnemonic,
+    bip32::Mnemonic,
```
Comment (GhostOfGauss) [inline manta-pay/src/key.rs:236]: Docs still say "BIP39 Mnemonic"

File: manta-pay/src/key.rs [hunk 2]
```diff
@@ -240,28 +233,26 @@ where
...
 );
 
-/// Seed Type
-pub type Seed = [u8; 64];
-
 impl Mnemonic {
-    /// Create a new BIP39 mnemonic phrase from the given phrase.
+    /// Create a new BIP39 mnemonic phrase from the given string.
```
Comment (GhostOfGauss) [inline manta-pay/src/key.rs:240]:
```suggestion
    /// Create a new BIP32 mnemonic phrase from the given string.
```

File: manta-pay/src/key.rs [hunk 3]
```diff
@@ -273,7 +264,7 @@ impl Mnemonic {
     /// Serializes the underlying `mnemonic` phrase.
     #[cfg(feature = "serde")]
     #[inline]
-    fn serialize<S>(mnemonic: &bip0039::Mnemonic, serializer: S) -> Result<S::Ok, S::Error>
+    fn serialize<S>(mnemonic: &bip32::Mnemonic, serializer: S) -> Result<S::Ok, S::Error>
```
Comment (GhostOfGauss) [inline manta-pay/src/key.rs:267]: Docs still say "BIP39 seed"

**Review by GhostOfGauss — APPROVED** [review]

File: manta-pay/src/key.rs [hunk 4]
```diff
@@ -122,33 +119,27 @@ impl_coin_type!(
...
 /// Key Secret
 #[cfg_attr(
     feature = "serde",
     derive(Deserialize, Serialize),
-    serde(crate = "manta_util::serde", deny_unknown_fields)
+    serde(crate = "manta_util::serde", deny_unknown_fields, transparent)
 )]
 #[derive(derivative::Derivative)]
-#[derivative(Clone(bound = ""))]
+#[derivative(Clone(bound = ""), Default(bound = ""))]
```
[the remark this answers is not in the document]
Comment (SupremoUGH) [reply manta-pay/src/key.rs]: Now it's gone because Mnemonic doesn't.

**Review by GhostOfGauss — DISMISSED** [review]

**Review by GhostOfGauss — APPROVED** [review]

**Review by bhgomes — APPROVED** [review]

**Pull request merged**